test/Microsoft.ML.Core.Tests/UnitTests/TestUtilities.cs

```diff
@@ -0,0 +1,169 @@
+            var x1Temp = x[1];
+            var x7Temp = x[7];
+            x[1] = x[0];
+            x[7] = x[6];
+            Assert.True(Utils.IsSorted(x));
+            x[1] = x1Temp;
+            x[7] = x7Temp;
+
+            // Not sorted
+            x[1] = x[6];
+            Assert.False(Utils.IsSorted(x));
+            x[1] = x1Temp;
+        }
+
+        [Fact]
+        [TestCategory("Utilities")]
+        public void CheckIsSortedFloat()
+        {
+            // A sorted (increasing) array
+            List<float> x = Enumerable.Range(0, 1000000).Select(i => (float)i).ToList();
+            Assert.True(Utils.IsSorted(x));
+
+            // A monotonically increasing array
+            var x1Temp = x[1];
+            var x7Temp = x[7];
+            x[1] = x[0];
+            x[7] = x[6];
+            Assert.True(Utils.IsSorted(x));
+            x[1] = x1Temp;
+            x[7] = x7Temp;
+
+            // Not sorted
+            x[1] = x[6];
+            Assert.False(Utils.IsSorted(x));
+            x[1] = x1Temp;
+            
+            // NaN: `Array.Sort()` will put NaNs into the first position,
+            // but we want to guarantee that NaNs aren't allowed in these arrays.
+            var x0Temp = x[0];
+            x[0] = float.NaN;
+            Assert.False(Utils.IsSorted(x));
+            x[0] = x0Temp;
+        }
+
+        [Fact]
+        [TestCategory("Utilities")]
+        public void CheckIsIncreasing()
+        {
+            // An increasing array
+            int[] x = Enumerable.Range(0, 10).ToArray();
+            Assert.True(Utils.IsIncreasing(0, x, 10));
+            // Check the lower bound
+            Assert.False(Utils.IsIncreasing(1, x, 10));
+            // The upper bound should be exclusive
+            Assert.False(Utils.IsIncreasing(0, x, 9));
+            // Any len should work
+            Assert.True(Utils.IsIncreasing(0, x, 5, 10));
+
+            // A monotonically increasing array should fail
+            var x7Temp = x[7];
+            x[7] = x[6];
+            Assert.False(Utils.IsIncreasing(0, x, 10));
+            // But until then, it should be fine
+            Assert.True(Utils.IsIncreasing(0, x, 7, 10));
+            x[7] = x7Temp;
+
+            // Not sorted
+            x[7] = x[9];
+            Assert.False(Utils.IsIncreasing(0, x, 10));
+            // Before the mismatched entry, it should be fine
+            Assert.True(Utils.IsIncreasing(0, x, 7, 10));
+            x[1] = x7Temp;
+        }
+
+        [Fact]
+        [TestCategory("Utilities")]
+        public void CheckIsAreEqualInt()
+        {
+            // A sorted (increasing) array
+            int[] x = Enumerable.Range(0, 10).ToArray();
+            int[] y = Enumerable.Range(0, 10).ToArray();
+            Assert.True(Utils.AreEqual(x, y));
+
+            // Not Equal
+            var x1Temp = x[1];
+            x[1] = x[0];
+            Assert.False(Utils.AreEqual(x, y));
+            x[1] = x1Temp;
```

Comment:
Minor nit, each test you are resetting back to the original value but its at the end of the test so the reset isnt necessary.

Reply:
Yes. I debated this. I went for this to help out anybody who may want to test more conditions by adding onto this function.